packages/node/src/lib/log.ts

```diff
@@ -0,0 +1,145 @@
+import type { LogOptions } from './construct-payload';
+import type { GroupingObject, OutgoingLogBody } from './metrics-log';
+import type { Request, Response } from 'express';
+import config from '../config';
+import clamp from 'lodash/clamp';
+import * as url from 'url';
+import { v4 as uuidv4 } from 'uuid';
+import { constructPayload } from './construct-payload';
+import { metricsAPICall } from './metrics-log';
+
+let queue: OutgoingLogBody[] = [];
+function doSend(readmeApiKey, options) {
+  // Copy the queue so we can send all the requests in one batch
+  const json = queue.slice();
+  // Clear out the queue so we don't resend any data in the future
+  queue = [];
+
+  // Make the log call
+  metricsAPICall(readmeApiKey, json).catch(e => {
+    // Silently discard errors and timeouts.
```

Comment:
this always worries me. There should be some way for a user to get errors even if not in development mode, maybe add an option and log it as console.error?

Reply:
Yeah agreed. Tbh I really dislike the idea of a development mode, it's messy and just hides data currently in the dash (it needs much improvements). Can you think of any other API ideas to log errors when they happen? 

Maybe we could accept a `logger` option?

```js
readmeio.log(process.env.README_API_KEY, req, res, group, { logger: console });
```

Or we could use https://www.npmjs.com/package/debug?

Reply:
I like the logger option and also that debug package is great. We should really be enforcing something like `debug` in all our codebases cause atm there isn't a dominant convention.

Reply:
Yeahh the tough thing is DEBUG is node.js specific (afaik?) so we have to either figure out a similar pattern that can be applied to all languages OR find the dominant pattern per language and apply that.

Reply:
One idea that @erunion had at one point would be for us to attempt to figure out what env they're running in, and log stuff out if they're in development. In Node.js we could use NODE_ENV, in .net you can use `app.Environment.IsDevelopment()` etc etc.